heron/packing/src/java/com/twitter/heron/packing/roundrobin/RoundRobinPacking.java

```diff
@@ -343,4 +353,32 @@ private void validatePackingPlan(PackingPlan plan) throws PackingException {
       }
     }
   }
+
+  @Override
+  public PackingPlan repack(PackingPlan currentPackingPlan, Map<String, Integer> componentChanges)
```

Comment:
Please add comment about what the logic below is doing and why. This function is not very straightforward.

Also are there unit tests for this class?

Reply:
+1 for unit tests. Particularly, the unit tests should guarantee situations below not to happen:
- count of instances in a container grows boundlessly
- count of instances in a container reduces to 1 eventually

Reply:
added comment

Reply:
added unit test